x/fees/keeper/grpc_query.go

```diff
@@ -155,3 +147,52 @@ func (k Keeper) DeployerFees(
 		Pagination: pageRes,
 	}, nil
 }
+
+// WithdrawFees returns all fees for a given withdraw address
+func (k Keeper) WithdrawFees(
+	c context.Context,
+	req *types.QueryWithdrawFeesRequest,
+) (*types.QueryWithdrawFeesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if strings.TrimSpace(req.WithdrawAddress) == "" {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			"deployer address is empty",
+		)
+	}
+
+	deployer, err := sdk.AccAddressFromBech32(req.WithdrawAddress)
+	if err != nil {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"invalid format for withdraw addr %s, should be bech32 ('evmos...')", req.WithdrawAddress,
+		)
+	}
+
+	var fees []types.Fee
```

Comment:
this should only be a list of addresses IMO, instead of the full list of fees

Reply:
@fedekunze you mean that the withdrawFees and deployerFees endpoints respond with contract addresses?

Reply:
yeah, a list of `[]string` where each string corresponds to an `sdk.AccAddress`